src/components/Tooltip/Tooltip.spec.tsx

```diff
@@ -3,27 +3,145 @@
 import { mount } from "@cypress/react";
 import React from "react";
 import { Tooltip } from "./Tooltip";
+import IconIcons from "@elements/Icon/Generated/IconIcons";
+import { BrightHeaderStyle } from "./BrightHeader";
 
 const TOOLTIP_TEXT = "This is a tooltip";
-const TOOLTIP_ICON_ID = "[data-test-id=tooltip-icon]";
 const TOOLTIP_ID = "[data-test-id=tooltip]";
+const TOOLTIP_LINK_ID = "[data-test-id=tooltip-link]";
+const TOOLTIP_LINK_URL = "https://www.frontify.com";
+const TOOLTIP_HEADING_TEXT = "I'm a heading";
+const GENERIC_ICON = <IconIcons />;
+const BRIGHT_HEADER_ID = "[data-test-id=bright-header]";
 
 describe("Tooltip Component", () => {
     it("should render a tooltip", () => {
         mount(<Tooltip tooltip={TOOLTIP_TEXT} />);
 
-        cy.get(TOOLTIP_ICON_ID).should("be.visible");
-        cy.get(TOOLTIP_ICON_ID).trigger("mouseover");
-
         cy.get(TOOLTIP_ID).should("contain", TOOLTIP_TEXT);
     });
-    it("should focus with keyboard", () => {
+
+    it("should render an icon next to the tooltip", () => {
+        mount(<Tooltip tooltip={TOOLTIP_TEXT} tooltipIcon={GENERIC_ICON} />);
+
+        cy.get("svg[name=IconIcons]").should("be.visible");
+    });
+
+    it("should render a link", () => {
+        mount(<Tooltip tooltip={TOOLTIP_TEXT} linkUrl={TOOLTIP_LINK_URL} />);
+
+        cy.get(TOOLTIP_LINK_ID).should("be.visible").and("have.attr", "href").and("include", TOOLTIP_LINK_URL);
+    });
+
+    it("should render a link with custom text", () => {
+        const linkLabel = "Upgrade your plan.";
+
+        mount(<Tooltip tooltip={TOOLTIP_TEXT} linkUrl={TOOLTIP_LINK_URL} linkLabel={linkLabel} />);
+
+        cy.get(TOOLTIP_LINK_ID).should("be.visible").and("contain", linkLabel);
+    });
+
+    it("should not render the heading icon if the heading is missing", () => {
+        mount(<Tooltip tooltip={TOOLTIP_TEXT} headingIcon={GENERIC_ICON} />);
+
+        cy.get(TOOLTIP_ID).should("not.contain", GENERIC_ICON);
+    });
+
+    it("should render the heading and the heading icon when both are present", () => {
+        mount(<Tooltip tooltip={TOOLTIP_TEXT} heading={TOOLTIP_HEADING_TEXT} headingIcon={GENERIC_ICON} />);
+
+        cy.get(TOOLTIP_ID).should("contain.text", TOOLTIP_HEADING_TEXT);
+        cy.get("svg[name=IconIcons]").should("be.visible");
+    });
+
+    it("should not render a bright header by default", () => {
```

Comment:
I'd combine this with the first one and maybe call it _should render a default tooltip_

Reply:
You mean you would move the assertions to the default tooltip test and not specifically have a test for this use case?

Reply:
Yes exactly